src/permissions/Caveat.ts

```diff
@@ -256,7 +281,9 @@ export function decorateWithCaveats<
       throw new UnrecognizedCaveatTypeError(caveat.type);
     }
 
-    decorated = specification.decorator(decorated, caveat);
+    if (hasDecorator(specification)) {
+      decorated = specification.decorator(decorated, caveat);
+    }
```

Comment:
We should throw an error if we get here and `hasDecorator(specification) === false`. See main PR comment.